rusoto/core/src/signature.rs

```diff
@@ -513,6 +514,7 @@ fn build_hostname(service: &str, region: &Region) -> String {
                 }
                 Region::UsEast1 => "s3.amazonaws.com".to_string(),
                 Region::CnNorth1 => format!("s3.{}.amazonaws.com.cn", region.name()),
+                Region::CnNorthwest1 => format!("{}.{}.amazonaws.com.cn", service, region.name()),
```

Comment:
I *think* we can leave the "s3" bit in there and do an inline match on `CnNorth1` or `CnNorthwest1`.